src/commands/sourcemaps/upload.ts

```diff
@@ -141,47 +137,34 @@ export class UploadCommand extends Command {
   }
 
   // Fills the 'repository' field of each payload with data gathered using git.
-  private addRepositoryDataToPayloads = async (payloads: Payload[]) => {
+  private addRepositoryDataToPayloads = async (payloads: Sourcemap[]) => {
     const repositoryData = await getRepositoryData(await newSimpleGit(), this.context.stdout, this.repositoryURL)
     if (repositoryData === undefined) {
       return
     }
     await Promise.all(
       payloads.map(async (payload) => {
-        payload.gitRepositoryPayload = this.getRepositoryPayload(repositoryData, payload.sourcemapPath)
-        payload.gitRepositoryURL = repositoryData.remote
-        payload.gitCommitSha = repositoryData.hash
+        const repositoryPayload = this.getRepositoryPayload(repositoryData!, payload.sourcemapPath)
+        payload.addRepositoryData({
+          gitCommitSha: repositoryData.hash,
+          gitRepositoryPayload: repositoryPayload,
+          gitRepositoryURL: repositoryData.remote,
+        })
       })
     )
   }
 
-  private getApiHelper(): APIHelper {
-    if (!this.config.apiKey) {
-      throw new InvalidConfigurationError(`Missing ${chalk.bold('DATADOG_API_KEY')} in your environment.`)
-    }
-
-    return apiConstructor(getBaseIntakeUrl(), this.config.apiKey!)
-  }
-
   // Looks for the sourcemaps and minified files on disk and returns
   // the associated payloads.
-  private getMatchingSourcemapFiles = async (cliVersion: string): Promise<Payload[]> => {
+  private getMatchingSourcemapFiles = async (): Promise<Sourcemap[]> => {
     const sourcemapFiles = glob.sync(buildPath(this.basePath!, '**/*js.map'))
 
     return Promise.all(
       sourcemapFiles.map(async (sourcemapPath) => {
         const minifiedFilePath = getMinifiedFilePath(sourcemapPath)
         const minifiedURL = this.getMinifiedURL(minifiedFilePath)
 
-        return {
-          cliVersion,
-          minifiedFilePath,
-          minifiedUrl: minifiedURL,
-          projectPath: this.projectPath,
-          service: this.service!,
-          sourcemapPath,
-          version: this.releaseVersion!,
-        }
+        return new Sourcemap(minifiedFilePath, minifiedURL, sourcemapPath)
```

Comment:
`cliVersion`, `projectPath`, `service`, `version` are all common to all sourcemaps a single run, so no need to duplicate the info in all `Sourcemap` object.